insights/parsers/proc_limits.py

```diff
@@ -0,0 +1,120 @@
+"""
+ProcLimits - File ``/proc/<PID>/limits``
+========================================
+
+Parser for parsing the the ``limits`` file under special ``/proc/<PID>``
+directory.
+
+"""
+
+from .. import Parser, parser
+from ..parsers import parse_fixed_table
+
+HEADER_SUBSTITUTE = [('Soft Limit', 'Soft_Limit'), ('Hard Limit', 'Hard_Limit')]
+
+
+class ProcLimits(Parser):
+    """
+    Base class for parsing the ``limits`` file under special ``/proc/<PID>``
+    directory into a list of lines by using the ``parse_fixed_table`` function.
+
+    Each line is a dictionary of fields, named according to their definitions
+    in ``Limit``.
+
+    This class provides the '__len__' and '__iter__' methods to allow it to
+    be used as a list to iterate over the ``data`` data.
+
+    Each of the resource provided by this file will be set as an attribute and
+    the corresponding `Hard Limit` will be set as its value.
+
+    Typical content looks like::
+
+        Limit                     Soft Limit           Hard Limit           Units
+        Max cpu time              unlimited            unlimited            seconds
+        Max file size             unlimited            unlimited            bytes
+        Max data size             unlimited            unlimited            bytes
+        Max stack size            10485760             unlimited            bytes
+        Max core file size        0                    unlimited            bytes
+        Max resident set          unlimited            unlimited            bytes
+        Max processes             9                    99                   processes
+        Max open files            1024                 4096                 files
+        Max locked memory         65536                65536                bytes
+        Max address space         unlimited            unlimited            bytes
+        Max file locks            unlimited            unlimited            locks
+        Max pending signals       15211                15211                signals
+        Max msgqueue size         819200               819200               bytes
+        Max nice priority         0                    0
+        Max realtime priority     0                    0
+        Max realtime timeout      unlimited            unlimited            us
+
+    Examples:
+        >>> proc_limits = shared[ProcLimits]
+        >>> len(proc_limits)
+        16
+        >>> proc_limits.max_processes
+        '99'
+        >>> 'max_cpu_time' in proc_limits
+        True
+        >>> proc_limits.max_cpu_time
+        'unlimited'
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.max_cpu_time = ''
```

Comment:
This declaration stuff seems completely unnecessary, as I think the meanings of the particular limits aren't particularly relevant here.  I really applaud you setting properties on the object rather than relying simply on the data property.  To me it seems best to remove all the declarations from `__init__`, and just populate the attributes directly exactly as you've done in `parse_content`.

Reply:
I can also understand that you want the properties to be set even if the file is incomplete - but I'm not sure that's a likely scenario with files in `/proc`.  And if a new limit is introduced, this will be inaccurate as a listing of properties anyway.

Reply:
I would like to keep them `__init__` :). And will update the docsting in case there will be some new resources on different system/machine.

Reply:
OK, makes sense to me. will do it Thanks Paul